tests/conftest.py

```diff
@@ -272,13 +279,43 @@ def func(line, assert_exit_code=0, stdin=None, matcher=False):
 network calls recorded:
 {formatted_network_calls}"""
             raise Exception(message)
+        if match_out is not None:
+            _assert_matches(result.stdout, "stdout", match_out)
+        if match_err is not None:
+            _assert_matches(result.stderr, "stderr", match_err)
         if matcher:
             return result, OutputMatcher(result)
         return result
 
     return func
 
 
+def _assert_matches(text, text_name, match):
+    if isinstance(match, str) or isinstance(match, re.Pattern):
+        match = [match]
+    match = [_convert_match_tuple(m) for m in match]
+
+    compiled_matches = [
+        m if isinstance(m, re.Pattern) else re.compile(m, re.MULTILINE) for m in match
+    ]
+    for pattern in compiled_matches:
+        assert (
+            pattern.search(text) is not None
```

Comment:
`.search()` and `.match()` have different characteristics. Would it be sensible to call the parameters `search_stdout` and `search_stderr`?

Reply:
Yeah, I'm okay with using `search_*` to indicate that it's a regex search.